src/main/java/seedu/duke/storage/Storage.java

```diff
@@ -0,0 +1,100 @@
+
+    public static void saveModulesToFile(String filePath) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(filePath))) {
+            for (Module module : moduleList.getTakenModuleList()) {
+                writer.write(toString(module) + System.lineSeparator());
+            }
+        } catch (IOException e) {
+            System.out.println("An error occurred while saving modules to file: " + e.getMessage());
+        }
+    }
+
+    public static void ensureDirectoryExists(String filePath) {
+        File file = new File(filePath);
+        File parentDir = file.getParentFile();
+        if (!parentDir.exists()) {
+            parentDir.mkdirs();
+        }
+    }
+
+    public static void createFile(String filePath) {
+        ensureDirectoryExists(filePath);
+        try {
+            FileWriter file = new FileWriter(filePath);
+            file.close();
+        } catch (IOException e) {
+            System.out.println("An error occurred while creating file: " + e.getMessage());
+        }
+    }
+
+    public static void loadModulesFromFile() {
+        try {
+            String filePath = Paths.get(System.getProperty("user.dir"), "data", "moduleList.txt").toString();
+            File file = new File(filePath);
+            if (!file.exists()) {
+                createFile(filePath);
+            }
+            Scanner input = new Scanner(file);
+            while (input.hasNext()) {
+                String line = input.nextLine();
+                Module module = getModule(line);
+                moduleList.add(module);
+            }
+            input.close();
+        } catch (ModuleException | FileNotFoundException e) {
+            System.out.println("An error occurred while loading modules from file: " + e.getMessage());
+        }
+    }
+
+    private static Module getModule(String line) throws ModuleException {
+        try {
+            String[] parts = line.split(" ", 4);
+            String moduleCode = parts[0];
+            String moduleGrade = parts[1];
+            int moduleDate = Integer.parseInt(parts[2]);
+            String moduleStatus = parts[3];
+            if (jsonManager.moduleExist(moduleCode)) {
+                jsonManager.getModuleInfo(moduleCode);
+            } else {
+                throw new ModuleException("Module does not exist in NUS.");
+            }
+            int moduleMC = jsonManager.getModuleMC();
+            String moduleDescription = jsonManager.getModuleDescription();
+            Module module = new Module(moduleCode, moduleMC, moduleDate, moduleDescription);
+            if (moduleStatus.equals("true")) {
+                module.setModuleStatus(true);
+            }
+            if (!moduleGrade.equals("null")) {
+                module.setModuleGrade(moduleGrade);
+            }
+            return module;
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new ModuleException("Module data is corrupted.");
+        }
+    }
```

Comment:
Actually now that I think abt it, JsonManager could as well just have a method that can return `new Module(moduleCode, moduleMC, moduleDate, moduleDescription);`

But anyways, you made do with what u have.

Reply:
Yup we probably can iterate on it further, probably v2.1 to clean it up!